src/Symfony/Component/Serializer/Encoder/YamlDecode.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Encoder;
+
+use Symfony\Component\Yaml\Parser;
+
+class YamlDecode implements DecoderInterface
```

Comment:
I see no point to have both `YamlDecode`/`YamlEncode` and `YamlEncoder`. If user needs custom YAML encoder/decoder, he should create his `FooYamlEncoder implements EncoderInterface` and inject it.

The only case I see that user wants to replace all existing services with new implementation, but DIC has its own features for that, e.g. aliases.